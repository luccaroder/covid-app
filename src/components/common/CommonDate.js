import 'date-fns';
import React from 'react';
import DateFnsUtils from '@date-io/date-fns';
import {
  MuiPickersUtilsProvider,
  DatePicker,
} from '@material-ui/pickers';

export default function CommonDate({handleDateChange, selectedDate }) {

  return (
    <MuiPickersUtilsProvider utils={DateFnsUtils}>
        <DatePicker data-testid="common-date-test-id"
          autoOk
          variant="inline"
          format="dd/MM/yyyy"
          value={selectedDate}
          onChange={handleDateChange}
        />
    </MuiPickersUtilsProvider>
  );
}