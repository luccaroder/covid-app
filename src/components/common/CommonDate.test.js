import React from 'react';
import CommonDate from 'components/common/CommonDate'

const selectedDate = '03/03/2011'
afterEach(cleanup)

test('renders CommonDate react component', () => {
    const { asFragment } = render(<CommonDate selectedDate={selectedDate}/>);
    expect(asFragment(<CommonDate selectedDate={selectedDate}/>)).toMatchSnapshot();
});

test('CommonDate must be a correc property selectedDate', () => {
    const { container } = render(<CommonDate selectedDate={selectedDate}/>);
    expect(container.querySelector('input').value).toEqual(selectedDate);
});
