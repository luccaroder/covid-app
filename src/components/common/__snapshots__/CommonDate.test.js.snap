// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CommonDate react component 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root"
    data-testid="common-date-test-id"
    format="dd/MM/yyyy"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        readonly=""
        type="text"
        value="03/03/2011"
      />
    </div>
  </div>
</DocumentFragment>
`;
